activerecord/test/cases/finder_test.rb

```diff
@@ -1144,6 +1144,23 @@ def test_find_without_primary_key
       Matey.find(1)
     end
   end
+  
+  def test_first_and_last_with_limit_for_order_without_primary_key
+    # While Topic.first should impose an ordering by primary key,
+    # Topic.limit(n).first should not
+    assert_sql(/^((?!ORDER).)*$/) { Post.limit(1).first }
+    assert_sql(/^((?!ORDER).)*$/) { Post.limit(1).last }
```

Comment:
@schneems Okay, that reasoning makes sense. I’ve added a commit that removes the explicit SQL tests.

Worth noting that with the SQL tests gone, the remaining tests really only apply to Postgres, because SQLite and MySQL impose primary key ordering on their own by default.